packages/application-shell-connectors/src/components/application-context/application-context.js

```diff
@@ -134,9 +136,22 @@ const withApplicationContext = mapApplicationContextToProps => Component => {
   return WrappedComponent;
 };
 
+// eslint-disable-next-line import/no-mutable-exports
+let useApplicationContext;
```

Comment:
Instead of exporting undefined we could export a function throwing given there is no useContext with e.g: This is meant to be a forward compatible React hook to consume the application context. React hooks, more specifically useContext, do not seem to be available yet on the installed version of React.“.

Reply:
We can also get around `// eslint-disable-next-line import/no-mutable-exports` by doing

```js
const useApplicationContext = React.useContext
  ? mapApplicationContextToProps => {}
  : () => throw new Error('React hooks do not seem to be available yet on the installed version of React.') 
```